src/EnergyPlus/PlantPipingSystemsManager.cc

```diff
@@ -1093,7 +1093,11 @@ namespace PlantPipingSystemsManager {
             }
 
             // Farfield model
-            thisDomain.groundTempModel = GroundTemp::GetGroundTempModelAndInit(state, gtmType, s_ipsc->cAlphaArgs(3));
+            // Ok, this is a finicky bug, but I have to make a copy here. GetGroundTempModelAndInit takes name (last param) by const ref&
+            // It then calls FiniteDiffGroundTempsModel::FiniteDiffGTMFactory which also takes objectName by const ref&
+            // But it calls getObjectItem with s_ipsc->cAlphaArgs which overrides it, then the comparison fails
+            std::string const groundTempModelName = s_ipsc->cAlphaArgs(3);
+            thisDomain.groundTempModel = GroundTemp::GetGroundTempModelAndInit(state, gtmType, groundTempModelName);
```

Comment:
The fix is here. Take a copy.

Reply:
Yep, these DataIPShortCuts arrays can definitely be abused.  They were incredibly valuable in Fortran when we had to pre-size array arguments to pass to functions.  With vectors, it's just unnecessary.  Probably time to deprecate that stuff and just use local variables.

Reply:
Should we open an issue for this? I haven't looked if there's one (on a mobile device right now)